test/integration/targets/prepare_vmware_tests/tasks/main.yml

```diff
@@ -0,0 +1,104 @@
+---
+
+- when: vcsim is defined
+  block:
+  - name: start vcsim
+    uri:
+      url: http://{{ vcsim }}:5000/spawn?cluster=2
+    register: vcsim_instance
+  
+  - set_fact:
+      vcenter_hostname: "{{ vcsim }}"
+      vcenter_username: "{{ vcsim_instance.json.username }}"
+      vcenter_password: "{{ vcsim_instance.json.password }}"
+  
+  - debug:
+      var: vcsim_instance
+  
+  - name: get a list of hosts from vcsim
+    uri:
+      url: http://{{ vcsim }}:5000/govc_find?filter=H
+    register: hosts
+  
+  - name: get a host
+    set_fact:
+      host1: "{{ hosts.json[0] | basename }}"
+  
+  - debug: var=host1
+  
+  - name: get a list of Cluster from vcsim
+    uri:
+      url: http://{{ vcsim }}:5000/govc_find?filter=CCR
+    register: clusters
+  
+  - name: get a cluster
+    set_fact:
+      ccr1: "{{ clusters.json[0] | basename }}"
+  
+  - debug: var=ccr1
+  
+  - name: get a list of datacenter
+    uri:
+      url: http://{{ vcsim }}:5000/govc_find?filter=DC
+    register: datacenters
+  
+  - name: get a datacenter
+    set_fact:
+      dc1: "{{ datacenters.json[0] | basename }}"
+  
+  - debug: var=dc1
+  
+  - name: get a list of Datastores from vcsim
+    uri:
+      url: http://{{ vcsim }}:5000/govc_find?filter=D
+    register: datastores
+  - debug: var=datastores
+  
+  - set_fact:
+      ds1: "{{ datastores['json'][0]|basename }}"
+  - debug: var=ds1
+  
+  - name: get a list of distributed vswitch from vcsim after adding
+    uri:
+      url: "{{ 'http://' + vcsim + ':5000/govc_find?filter=DVS' }}"
```

Comment:
```suggestion
      url: http://{{ vcsim }}:5000/govc_find?filter=DVS
```

Reply:
Good point. I applied your suggestion on my local branch. I will push it later.